src/managedwriter/writer_client.ts

```diff
@@ -0,0 +1,311 @@
+    this._parent = parent ? parent : 'Please set a parent path';
+    this._client = new BigQueryWriteClient(bqWriteClientOpts) || client;
+    this._writeStreamType = writeStreamType || this._writeStreamType;
+    this._connections = {
+      connection_list: [],
+      connections: {},
+    };
+    this._streamId = 'Please open a connection to set connection name';
+    this._client_closed = false;
+  }
+
+  setParent = (projectId: string, datasetId: string, tableId: string): void => {
+    const parent = `projects/${projectId}/datasets/${datasetId}/tables/${tableId}`;
+    this._parent = parent;
+  };
+
+  getParent = (): string => {
+    return this._parent;
+  };
+
+  getStreamId = (): string => {
+    return this._streamId;
+  };
+
+  setStreamId = (streamId: string): void => {
+    this._streamId = streamId;
+  };
+
+  getClient = (): BigQueryWriteClient => {
+    return this._client;
+  };
+
+  setClient = (client: BigQueryWriteClient): void => {
+    this._client = client;
+  };
+
+  getWriteStreams = (writeStream: WriteStream): undefined | null | string[] => {
+    if (writeStream === undefined || writeStream.name === undefined) {
+      return undefined;
+    }
+    if (writeStream.name === null) {
+      return null;
+    }
+    return new Array(writeStream.name);
+  };
+
+  setWriteStreamType(streamType: WriteStream['type']): void {
+    this._writeStreamType = streamType;
+  }
+
+  getWriteStreamType(): WriteStream['type'] {
+    return this._writeStreamType;
+  }
+
+  getConnections(): StreamConnections {
+    return this._connections;
+  }
+
+  getClientClosedStatus(): boolean {
+    return this._client_closed;
+  }
+
+  async initializeStreamConnection(clientOptions?: CallOptions): Promise<void> {
+    if (this._client_closed) {
+      this._client_closed = false;
+    }
+    const request: protos.google.cloud.bigquery.storage.v1.ICreateWriteStreamRequest =
+      {
+        parent: this.getParent(),
+        writeStream: {type: this.getWriteStreamType()},
+      };
+    const [response] = await this._client.createWriteStream(request);
+    if (typeof [response] === undefined) {
+      throw new gax.GoogleError(`${response}`);
+    }
+    console.log(`Stream connection created: ${response.name}`);
+    try {
+      if (response.name) {
+        this.setStreamId(response.name);
+        const write_stream: WriteStream = {
+          name: response.name,
+          type: this.getWriteStreamType(),
+        };
+        const stream_connection: StreamConnection = {
+          write_stream: write_stream,
+          connection: clientOptions
+            ? this._client.appendRows(clientOptions)
+            : this._client.appendRows(),
+        };
+        this._connections.connection_list.push(stream_connection);
+        this._connections.connections[`${write_stream.name}`] =
+          stream_connection.connection;
+      }
+    } catch {
+      throw new Error('Stream connection failed');
+    }
+  }
+
+  async appendRowsToStream(
```

Comment:
This method would be part of a `ManagedStream` class, that will keep the connection open until we finish sending rows. The issue right now with this function is that is not reusing the connection that much, because is closing the stream when we get the first response.

Reply:
This method can potentially have a behavior similar to a `Promise`, where users can call `appendRows`, returning a "pending write" promise. This way users can call this multiple times and wait for multiple pending writes. Internally the ManagedStream will reuse the same connection and handle them in sequence and mark the pending write as complete. I'll add an example of that on the tests.